### PR TITLE
Fix tests with Vault Community Edition

### PR DESCRIPTION
**Pull request by remilapeyre**
This patch adds `TestEntPreCheck()` to some tests that were using namespaces or managed-keys. This fixes running the tests with Vault Community Edition which will make it easier for others to contribute.

It also adds the latest version of Vault CE in the CI to detect if something breaks in the future.

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

### CODE REVIEW

**Comment by fairclothjm** [comment]
@remilapeyre Thanks for the contribution! I think this makes sense.

This PR has a number of failing tests for the 1.15.0 community addition. I am not sure if you can view the logs [here](https://pipelinesghubeus23.actions.githubusercontent.com/iE1kCPQ7fyx3B2rDvN1onIUJn83kEspUhz6VexQVIzxw7FYh3e/_apis/pipelines/1/runs/9971/signedlogcontent/25?urlExpires=2023-10-06T14%3A25%3A29.3403686Z&urlSigningMethod=HMACV1&urlSignature=4bSpOJJMCXdl1BMYeCg0NFFALYhkQvwm8ITGGbkngzc%3D)? Looks like we need more Ent checks

**Comment by fairclothjm** [comment]
I think that link expired.


```
$ rg -o -N "FAIL: Test\w*" 6_acceptance\ \(vault1.15.0\).txt |uniq
FAIL: TestDecodeBasic
FAIL: TestDecodeBatch
FAIL: TestEncodeBasic
FAIL: TestEncodeBatch
FAIL: TestAlphabetName
FAIL: TestRoleName
FAIL: TestTemplateName
FAIL: TestTransformationName
FAIL: TestNewProviderMeta
FAIL: TestNewProviderMeta_Cert
FAIL: TestAccNamespaceProviderConfigure
FAIL: TestAccAppRoleAuthBackendRoleSecretID_wrapped_namespace
FAIL: TestAccAppRoleAuthBackendRoleSecretID_wrapped_namespace_withWrappedAccessor
FAIL: TestAccEndpointGoverningPolicy
FAIL: TestResourceGenericSecretNS
FAIL: TestAccGithubAuthBackend_ns
FAIL: TestAccKMIPSecretBackend_basic
FAIL: TestAccKMIPSecretBackend_remount
FAIL: TestAccKMIPSecretRole_basic
FAIL: TestAccKMIPSecretRole_remount
FAIL: TestAccKMIPSecretScope_remount
FAIL: TestManagedKeys
FAIL: TestMFADuoBasic
FAIL: TestMFAOktaBasic
FAIL: TestMFAPingIDBasic
FAIL: TestMFATOTPBasic
FAIL: TestResourceMountMangedKeys
FAIL: TestAccNamespace
FAIL: TestPkiSecretBackendCrlConfig
FAIL: TestQuotaLeaseCount
FAIL: TestQuotaLeaseCountWithRole
FAIL: TestAccRoleGoverningPolicy
FAIL: TestAccJWTAuthBackend_OIDC
FAIL: TestAccJWTAuthBackendProviderConfig
FAIL: TestAccJWTAuthBackend
```

**Review by benashz — COMMENTED** [review]
Good catch on this! I had some suggestions for you. Thanks.

**Comment by remilapeyre** [comment]
> I think that link expired.
> 
> ```
> $ rg -o -N "FAIL: Test\w*" 6_acceptance\ \(vault1.15.0\).txt |uniq
> FAIL: TestDecodeBasic
> FAIL: TestDecodeBatch
> FAIL: TestEncodeBasic
> FAIL: TestEncodeBatch
> FAIL: TestAlphabetName
> FAIL: TestRoleName
> FAIL: TestTemplateName
> FAIL: TestTransformationName
> FAIL: TestNewProviderMeta
> FAIL: TestNewProviderMeta_Cert
> FAIL: TestAccNamespaceProviderConfigure
> FAIL: TestAccAppRoleAuthBackendRoleSecretID_wrapped_namespace
> FAIL: TestAccAppRoleAuthBackendRoleSecretID_wrapped_namespace_withWrappedAccessor
> FAIL: TestAccEndpointGoverningPolicy
> FAIL: TestResourceGenericSecretNS
> FAIL: TestAccGithubAuthBackend_ns
> FAIL: TestAccKMIPSecretBackend_basic
> FAIL: TestAccKMIPSecretBackend_remount
> FAIL: TestAccKMIPSecretRole_basic
> FAIL: TestAccKMIPSecretRole_remount
> FAIL: TestAccKMIPSecretScope_remount
> FAIL: TestManagedKeys
> FAIL: TestMFADuoBasic
> FAIL: TestMFAOktaBasic
> FAIL: TestMFAPingIDBasic
> FAIL: TestMFATOTPBasic
> FAIL: TestResourceMountMangedKeys
> FAIL: TestAccNamespace
> FAIL: TestPkiSecretBackendCrlConfig
> FAIL: TestQuotaLeaseCount
> FAIL: TestQuotaLeaseCountWithRole
> FAIL: TestAccRoleGoverningPolicy
> FAIL: TestAccJWTAuthBackend_OIDC
> FAIL: TestAccJWTAuthBackendProviderConfig
> FAIL: TestAccJWTAuthBackend
> ```

Thanks for the list, there is so many tests, and some that depends on setting the correct Docker environment that I couldn't get a full list.
I will get to the rest of them during the week so that we can have a clean state with all the tests passing on CE.

**Comment by remilapeyre** [comment]
> I think that link expired.
> 
> ```
> $ rg -o -N "FAIL: Test\w*" 6_acceptance\ \(vault1.15.0\).txt |uniq
> FAIL: TestDecodeBasic
> FAIL: TestDecodeBatch
> FAIL: TestEncodeBasic
> FAIL: TestEncodeBatch
> FAIL: TestAlphabetName
> FAIL: TestRoleName
> FAIL: TestTemplateName
> FAIL: TestTransformationName
> FAIL: TestNewProviderMeta
> FAIL: TestNewProviderMeta_Cert
> FAIL: TestAccNamespaceProviderConfigure
> FAIL: TestAccAppRoleAuthBackendRoleSecretID_wrapped_namespace
> FAIL: TestAccAppRoleAuthBackendRoleSecretID_wrapped_namespace_withWrappedAccessor
> FAIL: TestAccEndpointGoverningPolicy
> FAIL: TestResourceGenericSecretNS
> FAIL: TestAccGithubAuthBackend_ns
> FAIL: TestAccKMIPSecretBackend_basic
> FAIL: TestAccKMIPSecretBackend_remount
> FAIL: TestAccKMIPSecretRole_basic
> FAIL: TestAccKMIPSecretRole_remount
> FAIL: TestAccKMIPSecretScope_remount
> FAIL: TestManagedKeys
> FAIL: TestMFADuoBasic
> FAIL: TestMFAOktaBasic
> FAIL: TestMFAPingIDBasic
> FAIL: TestMFATOTPBasic
> FAIL: TestResourceMountMangedKeys
> FAIL: TestAccNamespace
> FAIL: TestPkiSecretBackendCrlConfig
> FAIL: TestQuotaLeaseCount
> FAIL: TestQuotaLeaseCountWithRole
> FAIL: TestAccRoleGoverningPolicy
> FAIL: TestAccJWTAuthBackend_OIDC
> FAIL: TestAccJWTAuthBackendProviderConfig
> FAIL: TestAccJWTAuthBackend
> ```

Hi @fairclothjm, I had just forgotten to use the appropriate target for the tests in the CI. Can you approve the run so we can see if the tests are now passing?

**Comment by fairclothjm** [comment]
@remilapeyre Thanks! I just triggered the build to run. Looks like there are some merge conflicts currently

**Comment by fairclothjm** [comment]
Looks like all the tests are passing! 🎉

**Comment by remilapeyre** [comment]
Excellent, with all the tests passing, I will take care of @benashz review and fix the conflicts later today :)

File: vault/resource_pki_secret_backend_crl_config_test.go [hunk 1]
```diff
@@ -131,6 +132,10 @@ func setupCRLConfigTest(t *testing.T, preCheck func(), ignoreImportFields ...str
 			Check:  getCRLConfigChecks(resourceName, false),
 		},
 		{
+			SkipFunc: func() (bool, error) {
+				_, found := os.LookupEnv(testutil.EnvVarTfAccEnt)
+				return !found, nil
```
[the remark this answers is not in the document]
Comment (remilapeyre) [reply vault/resource_pki_secret_backend_crl_config_test.go]: Thanks!

**Comment by remilapeyre** [comment]
@fairclothjm everything should be good now!

**Comment by remilapeyre** [comment]
@fairclothjm Is this ready to go?

**Review by vinay-gopalan — APPROVED** [review]
Changes look good and tests are passing against both enterprise and CE versions of Vault. Thanks for adding these changes and contributing to the Terraform Vault Provider @remilapeyre! 🙏🏼

**Pull request merged**